jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -70,14 +78,59 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
         spec_key = list(current_spec.keys())[0]
         current_unit = current_spec[spec_key].spectral_axis.unit
     with app.data_collection.delay_link_manager_update():
+
+        # these are used to build a combined spectrum with all
+        # input spectra included (taken from https://github.com/spacetelescope/
+        # dat_pyinthesky/blob/main/jdat_notebooks/MRS_Mstar_analysis/
+        # JWST_Mstar_dataAnalysis_analysis.ipynb)
+        wlallorig = []
+        fnuallorig = []
+        dfnuallorig = []
+
         for i in range(len(data)):
             spec = data[i]
```

Comment:
```suggestion
        for i, spec in enumerate(data):
```

(could also include data_label with a zip, but I'm not sure that's necessary since you only use it in one place and only when `i==0`)

Reply:
I believe it would be out of scope for the current ticket. The basic requirement was just to read a spectrum list and plot the combination of all elements as a single plot. We can double check with @orifox. And yes, I see that there might be the need to add more user controls to select in between different possibilities for the load-and-display operation. But I believe the one we have right now is likely to be the default behavior anyways.